haystack_experimental/dataclasses/chat_message.py

```diff
@@ -67,6 +69,98 @@ def __repr__(self) -> str:
         fields_str = ", ".join(fields)
         return f"{self.__class__.__name__}({fields_str})"
 
+    @classmethod
+    def from_file_path(
+        cls,
+        file_path: Union[str, Path],
+        *,
+        size: Optional[Tuple[int, int]] = None,
+        mime_type: Optional[str] = None,
+        detail: Optional[Literal["auto", "high", "low"]] = None,
+        meta: Optional[Dict[str, Any]] = None,
+    ) -> "ImageContent":
+        """
+        Create an ImageContent object from a file path.
+
+        It exposes similar functionality as the `ImageFileToImageContent` component. For PDF to ImageContent conversion,
+        use the `PDFToImageContent` component.
+
+        :param file_path:
+            The path to the image file.
+        :param size:
+            If provided, resizes the image to fit within the specified dimensions (width, height) while
+            maintaining aspect ratio. This reduces file size, memory usage, and processing time, which is beneficial
+            when working with models that have resolution constraints or when transmitting images to remote services.
+        :param mime_type:
+            Optional MIME type of the image. If not provided, the MIME type is guessed from the file extension.
+        :param detail:
+            Optional detail level of the image (only supported by OpenAI). One of "auto", "high", or "low".
+        :param meta:
+            Additional metadata for the image.
+
+        :returns:
+            An ImageContent object.
+        """
+        # to avoid a circular import
+        from haystack_experimental.components.image_converters import ImageFileToImageContent
+
+        if mime_type:
+            source = ByteStream.from_file_path(Path(file_path), mime_type=mime_type, meta={"file_path": str(file_path)})
+        else:
+            source = file_path
+
+        converter = ImageFileToImageContent(size=size, detail=detail)
+        result = converter.run(sources=[source], meta=[meta] if meta else None)
+        return result["image_contents"][0]
+
+
+    @classmethod
+    def from_url(
+        cls,
+        url: str,
+        *,
+        retry_attempts: int = 2,
+        timeout: int = 10,
+        size: Optional[Tuple[int, int]] = None,
+        mime_type: Optional[str] = None,
+        detail: Optional[Literal["auto", "high", "low"]] = None,
+        meta: Optional[Dict[str, Any]] = None,
+    ) -> "ImageContent":
+        """
+        Create an ImageContent object from a URL.
```

Comment:
I'd recommend a slight rewording to make it clear we download the image from the URL, otherwise it might sound like we just store the URL in the ImageContent object.

Reply:
Makes sense. Done in https://github.com/deepset-ai/haystack-experimental/pull/294/commits/c955b14364deceaf997d1ca7da8e66f16cef9332